src/components/coins-page-table/CoinTable/CoinTable.js

```diff
@@ -21,92 +22,98 @@ import {
   THLast7d,
 } from "./CoinTable.styles";
 import { LoadingTableRow } from "components/loading-animations";
+import { useWindowSize } from "hooks";
 
 const CoinTable = (props) => {
-  const isLoading = useSelector(state => state.coins.isLoading)
-  const loaders = Array.apply(null, Array(20)).map( function() {})
-  const activeCurrency = useSelector(getActiveCurrency)
-  const activeCategory = useSelector(state => state.coins.activeCategory)
-  const activeDirection = useSelector(state => state.coins.activeDirection)
-  const activePageNum = useSelector(state => state.coins.pageNum)
-  const activeResultsPerPage = useSelector(state => state.coins.activeResultsPerPage)
+  const isLoading = useSelector((state) => state.coins.isLoading);
+  const { width: screenWidth } = useWindowSize();
+  const loaders = Array.apply(null, Array(20)).map(function () {});
+  const activeCurrency = useSelector(getActiveCurrency);
+  const activeCategory = useSelector((state) => state.coins.activeCategory);
+  const activeDirection = useSelector((state) => state.coins.activeDirection);
+  const activePageNum = useSelector((state) => state.coins.pageNum);
+  const activeResultsPerPage = useSelector(
+    (state) => state.coins.activeResultsPerPage
+  );
 
   useEffect(() => {
-    props.getCoinsDataByCategory(activeCategory)
+    props.getCoinsDataByCategory(activeCategory);
   }, []);
 
   useEffect(() => {
-    props.getCoinsDataByCategory(activeCategory)
-  }, [activeCurrency])
+    props.getCoinsDataByCategory(activeCategory);
+  }, [activeCurrency]);
 
   useEffect(() => {
-    props.getCoinsDataByCategory(activeCategory)
-  }, [activeCategory])
+    props.getCoinsDataByCategory(activeCategory);
+  }, [activeCategory]);
 
   useEffect(() => {
-    props.getCoinsDataByCategory(activeCategory)
-  }, [activeDirection])
+    props.getCoinsDataByCategory(activeCategory);
+  }, [activeDirection]);
 
   useEffect(() => {
-    props.getCoinsDataByCategory(activeCategory)
-  }, [activePageNum])
+    props.getCoinsDataByCategory(activeCategory);
+  }, [activePageNum]);
 
   useEffect(() => {
-    props.getCoinsDataByCategory(activeCategory)
-  }, [activeResultsPerPage])
+    props.getCoinsDataByCategory(activeCategory);
+  }, [activeResultsPerPage]);
```

Comment:
can combine all these use effects into one?

Reply:
I fix this in the next commit